### ISSUE TITLE
ArabicLexer does not store BeforeAnnotation and AfterAnnotation whitespace.

### ISSUE THREAD

**Issue by tc64** [stanfordnlp/CoreNLP#113]
First, thank you all for providing this resource. I would like to use your Arabic processing tools but am running into a problem: the tokenizer doesn't record the amount of whitespace before and after a token.

I see that in getNext() in ArabicLexer, the syntax that would set values for CoreAnnotations.BeforeAnnotation and CoreAnnotations.AfterAnnotation to be the appropriate amount of whitespace before or after a token is commented out. I tried uncommenting and added private CoreLabel prevWord and private StringBuilder prevWordAfter to ArabicLexer, and made sure invertible=true. I'm not sure why, but the BeforeAnnotation and AfterAnnotation values always end up as empty strings. prevWordAfter.StringBuilder.toString() is always returning an empty string. 

Could anyone point me in the right direction as to how to fix this problem?

Thank you.
-Taylor

**Comment by tc64** [comment]
For what its worth, here is a solution that works for my purposes...

To store Before and AfterAnnotation whitespace (i.e. properly use the "invertible" option), one can follow the pattern in PTBLexer. I did the following:

Initialize prevWordAfter and prevWord as in PTBLexer.
In getNext(String txt, String originalText), uncomment the lines 977, 978, 981, 982, 983.
In next(), edit the appropriate case(s) in the switch (zzAction < 0 ? zzAction : ZZ_ACTION[zzAction]) block to append whitespace to prevWordAfter. I added if (invertible) { prevWordAfter.append(yytext()) }; to case 13. PTBLexer has several cases where this conditional operation is performed.
